pkg/trace/api/debugger_test.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+)
+
+func TestDebuggerProxy(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		slurp, err := ioutil.ReadAll(req.Body)
+		_ = req.Body.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+		if body := string(slurp); body != "body" {
+			t.Fatalf("invalid request body: %q", body)
+		}
+		if v := req.Header.Get("DD-API-KEY"); v != "123" {
+			t.Fatalf("got invalid API key: %q", v)
+		}
+		if query := req.URL.RawQuery; query != "ddtags=key%3Aval" {
+			t.Fatalf("got invalid query params: %q", query)
+		}
+		_, err = w.Write([]byte("OK"))
+		if err != nil {
+			t.Fatal(err)
+		}
+	}))
+	u, err := url.Parse(srv.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req, err := http.NewRequest("POST", "dummy.com/path", strings.NewReader("body"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	rec := httptest.NewRecorder()
+	c := &traceconfig.AgentConfig{}
+	newDebuggerProxy(c.NewHTTPTransport(), u, "123", "key:val").ServeHTTP(rec, req)
+	slurp, err := ioutil.ReadAll(rec.Result().Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(slurp) != "OK" {
+		t.Fatal("did not proxy")
+	}
+}
+
+func testHandler(conf *traceconfig.AgentConfig, assertions []func(req *http.Request) error) (string, error) {
```

Comment:
I think you were supposed to delete this :)

Reply:
Ah, yep.